src/validation/custom.validation.ts

```diff
@@ -68,10 +68,14 @@ export const checkDateIsWithinLast3Months = (errMsg: string, day: string = "", m
   return true;
 };
 
+const digitsWithLeadingZero = ["01", "02", "03", "04", "05", "06", "07", "08", "09"];
+
 export const checkDateValueIsValid = (invalidDateErrMsg: string, dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
   const day = parseInt(dayStr), month = parseInt(monthStr), year = parseInt(yearStr);
   if (isNaN(day) || isNaN(month) || isNaN(year) ||
-    day.toString() !== dayStr || month.toString() !== monthStr || year.toString() !== yearStr ||
+    (day.toString() !== dayStr && !digitsWithLeadingZero.includes(dayStr)) ||
+    (month.toString() !== monthStr && !digitsWithLeadingZero.includes(monthStr)) ||
```

Comment:
This change works, although I think the use of regex would be simpler

```const day = parseInt(dayStr), month = parseInt(monthStr), year = parseInt(yearStr);
  const dayMonthRegex = /^\d{1,2}$/;
  const yearRegex = /^\d{4}$/;

  if (isNaN(day) || isNaN(month) || isNaN(year) ||
    !dayMonthRegex.test(dayStr) || !dayMonthRegex.test(monthStr) || !yearRegex.test(yearStr) ||
    !DateTime.utc(year, month, day).isValid) {
    throw new Error(invalidDateErrMsg);
  }
```

Reply:
used the regex for day/month but year 0999 was failing in unit test so have used the original year check.
This is also inline with our new story to allow leading zero in day and month - https://companieshouse.atlassian.net/browse/UAR-1357

Reply:
push didn't complete, will try again...